### PR TITLE
Fix temporary SQL rows leak in SqlFetchChecksums()

### LINKED ISSUE #146: Icingadb bombs mysql with connections
How to reproduce:

[root@p3i212t4-idb ~]# systemctl start icingadb; ss -lautnp | grep 3306 | grep 'TIME-WAIT' -c; sleep 10; ss -lautnp | grep 3306 | grep 'TIME-WAIT' -c
```
0
386
```
On database side we have a following picture:
```
[root@p3i212t4-d01 log]# ss -atunp | grep 3306
tcp    LISTEN     0      80        *:3306                  *:*                   users:(("mysqld",pid=11918,fd=37))
tcp    TIME-WAIT  0      0      10.32.140.99:3306               10.33.111.114:60494
tcp    ESTAB      0      0      10.32.140.99:3306               10.37.101.222:32794               users:(("mysqld",pid=11918,fd=63))
tcp    TIME-WAIT  0      0      10.32.140.99:3306               10.33.111.114:32848
tcp    TIME-WAIT  0      0      10.32.140.99:3306               10.33.111.114:32822
tcp    TIME-WAIT  0      0      10.32.140.99:3306               10.33.111.114:33228
tcp    TIME-WAIT  0      0      10.32.140.99:3306               10.33.111.114:33156
tcp    TIME-WAIT  0      0      10.32.140.99:3306               10.33.111.114:33294
```
connection stats from mysql:
```
MariaDB [(none)]> show variables like "max_connections";
+-----------------+-------+
| Variable_name   | Value |
+-----------------+-------+
| max_connections | 151   |
+-----------------+-------+
``` 

Environment:
- Standalone icinga2 master: 8 cores / 16 GB ram
- 2 Satellite zones in HA(4 icinga2 instances as satellites): 8 cores / 16 GB ram
- 1 IcingaDB + Redis on dedicated host: 8 cores / 16 GB ram
- MySQL cluster on dedicated hosts: 8 cores / 16 GB ram

CentOS Linux release 7.7.1908 (Core) on each server
Amount of active services: 102010 services per minute

What has been done:
I've played around with options in /etc/icingadb/icingadb.ini
max_open_conns option does not make any affect, i've tried following values: 1, 50(default), 512

Error could be caused by incorrect connection closure by icingadb daemon, please let me know if i can provide additional debug output for you

With current environment constantly have ~50k services in Overdue.

### PR DESCRIPTION
**Pull request by Al2Klimov**
fixes #146

### CODE REVIEW

File: connection/mysql.go [hunk 1]
```diff
@@ -680,6 +680,8 @@ func (dbw *DBWrapper) SqlFetchChecksums(table string, ids []string) (map[string]
 		if err != nil {
 			return nil, err
 		}
+
+		rows.Close()
```
Comment (lippserd) [inline connection/mysql.go:684]: There already is a `defer rows.Close()`. Why the extra call?

Comment (lippserd) [reply connection/mysql.go:684]: Because we're looping here, sry.

Comment (Al2Klimov) [reply connection/mysql.go:684]: To close the rows ASAP.

**Comment by Al2Klimov** [comment]
Rationale: https://github.com/go-sql-driver/mysql/issues/111#issuecomment-21433114

**Pull request merged**